src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -2353,5 +2353,30 @@ private int getArrayOrStringLength(CORINFO_OBJECT_STRUCT_* objHnd)
                 _ => -1
             };
         }
+
+        private bool getIsClassInitedFlagAddress(CORINFO_CLASS_STRUCT_* cls, ref CORINFO_CONST_LOOKUP addr, ref int offset)
+        {
+            MetadataType type = (MetadataType)HandleToObject(cls);
+            addr.addr = (void*)ObjectToHandle(_compilation.NodeFactory.TypeNonGCStaticsSymbol(type));
```

Comment:
just to understand - this should not be `TypeGCStaticsSymbol`?

Reply:
Yes. There is only one initialization state for a given type and it is stored alongside non-GC statics in NAOT.